client/question/components/LearnerViewDirective.js

```diff
@@ -1046,6 +1046,7 @@ tie.directive('learnerView', [function() {
          */
         $scope.revertToSelectedSnapshot = function(selectedSnapshotIndex) {
           $scope.currentSnapshotIndex = selectedSnapshotIndex;
+          EventHandlerService.createCodeRestoreEvent();
```

Comment:
You may want to hold until you successfully retrieve the Snapshot, in case there's an error.